packages/core/src/shared/utilities/functionUtils.ts

```diff
@@ -156,3 +157,43 @@ export function keyedDebounce<T, U extends any[], K extends string = string>(
         return promise
     }
 }
+/**
+ * Creates a function that runs only for unique arguments that haven't been seen before.
+ *
+ * This utility tracks all unique inputs it has seen and only executes the callback
+ * for new inputs. Unlike `onceChanged` which only compares with the previous invocation,
+ * this function maintains a history of all arguments it has processed.
+ *
+ * @param fn The function to execute for unique arguments
+ * @param key A function that returns a unique string for each argument set, defaults to joining with ":" seperating
+ * @param overflow The maximum number of unique arguments to store.
+ * @returns A wrapped function that only executes for new unique arguments
+ *
+ * @example
+ * ```ts
+ * const logOnce = oncePerUniqueArg((message) => console.log(message))
+ *
+ * logOnce('hello') // prints: hello
+ * logOnce('world') // prints: world
+ * logOnce('hello') // nothing happens (already seen)
+ * logOnce('test')  // prints: test
+ * ```
+ */
+export function oncePerUniqueArg<T, U extends any[]>(
```

Comment:
off-topic: Might be able to drop `onceChanged` in favor of this.

Reply:
Yeah its somewhat of a generalization. I think we could reimplement `onceChanged` to call `oncePerUniqueArg` with a buffer size of 1 so that it only compares against the previous.